### PR TITLE
cleaned up parity/signer

### PR DESCRIPTION
**Pull request by debris**
- cleaned up parity/signer
- fixed compiling parity without `ethcore-signer` feature

### CODE REVIEW

File: parity/signer/off.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::io;
+use die::*;
+use super::{Configuration, Dependencies};
+
+pub struct SignerServer;
+
+pub fn start(_conf: Configuration, _deps: Dependencies) -> ! {
```
Comment (tomusdrw) [inline parity/signer/off.rs]: So the problem with this approach is that I'm pretty sure that file will be never modified along with `on.rs` (you will simply forget that you should change something here - cause we never compile without the feature).

That said, I like that approach - much cleaner, although will be difficult to maintain.

**Comment by gavofyork** [comment]
why do we have signer as optional still?

**Comment by tomusdrw** [comment]
It's just compilation feature - like `rpc`, `dapps` and `jit` although some of them are enabled by default. It has to be feature cause it's dependent on `rpc`

**Comment by gavofyork** [comment]
use case for rpc being optional?

**Comment by gavofyork** [comment]
it seems there isn't one - i suggest this is refactored to make rpc and signer compiled in.

**Comment by debris** [comment]
rpc and signer are no longer compile time options and are always compiled in

**Pull request merged**